src/radical/pilot/agent/scheduler/base.py

```diff
@@ -450,11 +453,14 @@ def _change_slot_states(self, slots, new_state):
     #
     # NOTE: any scheduler implementation which uses a different nodelist
     #       structure MUST overload this method.
-    def slot_status(self, msg=None):
+    def slot_status(self, msg=None, uid=None):
         '''
         Returns a multi-line string corresponding to the status of the node list
         '''
 
+        if not _debug:
+            return
+
         if not self._log.isEnabledFor(logging.DEBUG):
             return
```

Comment:
we shouldn't have such internal flag `_debug`, and we do have a check for debug mode (`if not self._log.isEnabledFor(logging.DEBUG):`)

Reply:
adding to the above reply: even if we have `debug` enabled, the slot status is usually too costly to include, thus the separate flag.

Reply:
This has been addressed - the `debug_level` of the logger is checked now.